pkg/agent/controller/egress/egress_controller.go

```diff
@@ -231,6 +256,24 @@ func (c *EgressController) Run(stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+func (c *EgressController) shouldStartCluster() bool {
+	ipPools, _ := c.ipPoolInformer.Lister().List(labels.Everything())
+	for _, ip := range ipPools {
+		nodeSel := ip.Spec.NodeSelector
+		if nodeSel == nil {
+			return true
+		}
+		nodeSelect, _ := metav1.LabelSelectorAsSelector(nodeSel)
+		nodes, _ := c.nodeInformer.Lister().List(nodeSelect)
```

Comment:
Getting the Node's labels and check it with the nodeSelector should be more efficient, compared with iterating all Nodes twice.

Reply:
Add nodeSelectors (map[string]*metav1.LabelSelector) which hold all node selectors in Cluster struct

Reply:
done